assets/cluster-monitoring-operator/prometheus-rule.yaml

```diff
@@ -32,15 +32,16 @@ spec:
       annotations:
         description: Workload {{ $labels.namespace }}/{{ $labels.workload }} is incorrectly
           spread across multiple nodes which breaks high-availability requirements.
-          There are {{ $value }} pods on node {{ $labels.node }}, where there should
-          only be one. Since the workload is using persistent volumes, manual intervention
-          is needed. Please follow the guidelines provided in the runbook of this
-          alert to fix this issue.
+          Since the workload is using persistent volumes, manual intervention is needed.
+          Please follow the guidelines provided in the runbook of this alert to fix
+          this issue.
         runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/HighlyAvailableWorkloadIncorrectlySpread.md
         summary: Highly-available workload is incorrectly spread across multiple nodes
           and manual intervention is needed.
       expr: |
-        count by(node,workload,namespace)
+        count without (node)
+        (
+          group by (node, workload, namespace)
```

Comment:
Syntax-wise the aggregation is different.

```
count without (node) (
  group by (node, workload, namespace) ( ... )
)
```
The inner aggregation is here to aggregate away all the labels that don't matter so that they won't impact the final count. And the outer aggregation removes the `node` since we want to focus the count solely on the number of unique nodes that each namespace/workload pair is scheduled on.

If we were to use only one aggregation, we would be accounting for the number of unique label values that each namespace/workload pair has. For instance, if two workloads are scheduled on the same node, but one of their labels is different, let's say the pod name, they would count as 2 different series in the aggregation where we would want them to be one.